### PR TITLE
NestJs with monorepo serverless-architecture & CT Client service

### PR DESCRIPTION
**Pull request by Yash-D-composeTech**

### CODE REVIEW

File: nest-commercetools-project/apps/auth-app/src/auth-app.controller.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export class AuthAppController {
   constructor(private readonly authAppService: AuthAppService) {}
 
   @Get()
-  getHello(): string {
-    return this.authAppService.getHello();
+  getTotalCustomers(): Promise<object> {
```
Comment (0xakuma) [inline nest-commercetools-project/apps/auth-app/src/auth-app.controller.ts]: Can we avoid object type, we can use CT types

File: nest-commercetools-project/apps/auth-app/src/main.ts [hunk 2]
```diff
@@ -1,8 +1,42 @@
...
+    }),
+  );
+
+  await app.init();
+
+  // Create an express app
+  const expressApp = express();
+  expressApp.use('/auth-app', app.getHttpAdapter().getInstance()); // Mount the NestJS app on "/auth-app" route
+
+  // const expressApp = app.getHttpAdapter().getInstance();
```
Comment (0xakuma) [inline nest-commercetools-project/apps/auth-app/src/main.ts]: remove comments if not needed

File: nest-commercetools-project/apps/product-app/src/product-app.controller.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ export class ProductAppController {
   constructor(private readonly productAppService: ProductAppService) {}
 
   @Get()
-  getHello(): string {
-    return this.productAppService.getHello();
+  getTotalProducts(): Promise<object> {
```
Comment (0xakuma) [inline nest-commercetools-project/apps/product-app/src/product-app.controller.ts]: Use CT types

File: nest-commercetools-project/libs/common/src/ctClient/ctClient.services.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+  constructor() {
+    this.CT_AUTH_URL = process.env.CT_AUTH_URL;
+    this.CT_PROJECT_KEY = process.env.CT_PROJECT_KEY;
+    this.CT_API_URL = process.env.CT_API_URL;
+    this.CT_CLIENT_ID = process.env.CT_CLIENT_ID;
+    this.CT_CLIENT_SECRET = process.env.CT_CLIENT_SECRET;
+
+    // this.scope = `view_stores:${this.CT_PROJECT_KEY} manage_my_payments:${this.CT_PROJECT_KEY} manage_my_orders:${this.CT_PROJECT_KEY} view_tax_categories:${this.CT_PROJECT_KEY} create_anonymous_token:${this.CT_PROJECT_KEY} view_categories:${this.CT_PROJECT_KEY} manage_my_quotes:${this.CT_PROJECT_KEY} introspect_oauth_tokens:${this.CT_PROJECT_KEY} manage_my_shopping_lists:${this.CT_PROJECT_KEY} view_products:${this.CT_PROJECT_KEY} manage_my_quote_requests:${this.CT_PROJECT_KEY} view_types:${this.CT_PROJECT_KEY} manage_my_business_units:${this.CT_PROJECT_KEY} manage_my_profile:${this.CT_PROJECT_KEY}`;
+    this.scope = `manage_project:${this.CT_PROJECT_KEY}`;
```
Comment (0xakuma) [inline nest-commercetools-project/libs/common/src/ctClient/ctClient.services.ts:31]: Should we use this scope for CT client ?

**Review by 0xakuma — CHANGES_REQUESTED** [review]

File: nest-commercetools-project/libs/common/src/ctClient/ctClient.services.ts (hunk 4 above)
Comment (Yash-D-composeTech) [reply nest-commercetools-project/libs/common/src/ctClient/ctClient.services.ts:31]: In my CT demo account I have created a demo client with manage_project scope That's why declare that scope for CT client. In future we can update it as per requirement
image

**Review by 0xakuma — APPROVED** [review]

**Pull request merged**